thirdparty/github.go

```diff
@@ -357,42 +358,43 @@ func parseGithubErrorResponse(resp *github.Response) error {
 // GetGithubFile returns a struct that contains the contents of files within
 // a repository as Base64 encoded content. Ref should be the commit hash or branch (defaults to master).
 func GetGithubFile(ctx context.Context, token, owner, repo, path, ref string) (*github.RepositoryContent, error) {
-	installationToken, _ := GetInstallationToken(ctx, owner, repo, nil)
-
-	httpClient := getGithubClientRetry(installationToken, "GetGithubFile")
-	defer utility.PutHTTPClient(httpClient)
-	client := github.NewClient(httpClient)
-
 	var opt *github.RepositoryContentGetOptions
 	if len(ref) != 0 {
 		opt = &github.RepositoryContentGetOptions{
 			Ref: ref,
 		}
 	}
 
-	file, _, resp, err := client.Repositories.GetContents(ctx, owner, repo, path, opt)
-	if resp != nil {
-		defer resp.Body.Close()
-		if err == nil && file != nil && file.Content != nil {
-			return file, nil
+	installationToken, _ := GetInstallationToken(ctx, owner, repo, nil)
+	if installationToken != "" {
+		httpClient := getGithubClientRetry(installationToken, "GetGithubFile")
+		defer utility.PutHTTPClient(httpClient)
+		client := github.NewClient(httpClient)
+
+		file, _, resp, err := client.Repositories.GetContents(ctx, owner, repo, path, opt)
+		if resp != nil {
+			defer resp.Body.Close()
+			if err == nil && file != nil && file.Content != nil {
+				return file, nil
+			}
 		}
+
+		grip.Debug(message.WrapError(err, message.Fields{
```

Comment:
there's a chance we won't log this if we don't have an error but file or file content is nil; that being said, shouldn't we return the same error as below in that case? (like the call worked still)